tests/Insightly/InsightlyContactLinkTest.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Insightly;
+
+use App\Domain\Contacts\Contact;
+use App\Domain\Contacts\ContactType;
+use App\Insightly\InsightlyContactLink;
+use PHPUnit\Framework\TestCase;
+use Ramsey\Uuid\Uuid;
+use Tests\MockInsightlyClient;
+
+final class InsightlyContactLinkTest extends TestCase
+{
+    use MockInsightlyClient;
+
+    private InsightlyContactLink $contactLink;
+
+    protected function setUp(): void
+    {
+        $this->mockCrmClient();
+
+        $this->contactLink = new InsightlyContactLink($this->insightlyClient);
+    }
+
+    public function test_it_creates_a_new_contact_when_the_email_is_unique(): void
+    {
+        $email = 'info@publiq.be';
+        $insightlyId = 42;
+
+        $contact = $this->buildContactWithEmail($email);
```

Comment:
It seems that all the contacts have the same values inside this unit test, maybe it can become a property that gets created inside the `setUp` method?